### PR TITLE
build: update deps

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: [1.17, 1.18]
+        go: ['1.19', '1.20']
 
     services:
       postgres:
-        image: postgres:13.0
+        image: postgres:14.0
```
Comment (chappjc) [inline .github/workflows/go.yml:16]: 15?  It has been out since Oct 2022

File: go.mod [hunk 2]
```diff
@@ -3,50 +3,53 @@ module github.com/decred/dcrpool
...
-	github.com/decred/dcrd/chaincfg/chainhash v1.0.4-0.20210914212651-723d86274b0d
-	github.com/decred/dcrd/chaincfg/v3 v3.0.1-0.20210914212651-723d86274b0d
-	github.com/decred/dcrd/crypto/blake256 v1.0.1-0.20210914212651-723d86274b0d
-	github.com/decred/dcrd/dcrjson/v4 v4.0.0
-	github.com/decred/dcrd/dcrutil/v4 v4.0.0-20211103115647-1eff7272fddf
-	github.com/decred/dcrd/rpc/jsonrpc/types/v3 v3.0.0-20211103115647-1eff7272fddf
-	github.com/decred/dcrd/rpcclient/v7 v7.0.0-20211103115647-1eff7272fddf
-	github.com/decred/dcrd/txscript/v4 v4.0.0-20211103115647-1eff7272fddf
-	github.com/decred/dcrd/wire v1.4.1-0.20210914212651-723d86274b0d
+	decred.org/dcrwallet/v2 v2.2.0
```
Comment (chappjc) [inline go.mod:6]: probly want v3

File: run_tests.sh [hunk 3]
```diff
@@ -23,19 +23,5 @@ env GORACE="halt_on_error=1" go test -race ./...
 # golangci-lint (github.com/golangci/golangci-lint) is used to run each each
 # static checker.
 
-# set output format for linter
-if [[ -v CI ]]; then
-    OUT_FORMAT="github-actions"
-else
-    OUT_FORMAT="colored-line-number"
```
Comment (chappjc) [inline run_tests.sh:30]: These didn't make it to the yml file.  I also think you can specify both.

**Review by chappjc — CHANGES_REQUESTED** [review]

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -24,13 +27,13 @@ jobs:
           - 5432:5432
     steps:
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 #v4.0.1
         with:
           go-version: ${{ matrix.go }}
       - name: Check out source
-        uses: actions/checkout@v2
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab #v3.5.2
```
Comment (davecgh) [inline .github/workflows/go.yml:34]:
```suggestion
        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 #v3.5.3
```

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -24,13 +27,13 @@ jobs:
...
       - name: Install Linters
-        run: "curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.45.2"
+        run: "curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.53.2"
```
Comment (davecgh) [inline .github/workflows/go.yml:36]:
```suggestion
        run: "curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.53.3"
```

File: go.mod (hunk 2 above)
Comment (davecgh) [inline go.mod:6]:
```suggestion
	decred.org/dcrwallet/v3 v3.0.1
```

File: go.mod [hunk 6]
```diff
@@ -3,50 +3,53 @@ module github.com/decred/dcrpool
...
 	github.com/jessevdk/go-flags v1.4.1-0.20200711081900-c17162fe8fd7
 	github.com/jrick/logrotate v1.0.0
-	github.com/lib/pq v1.9.0
-	go.etcd.io/bbolt v1.3.5
-	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
-	golang.org/x/time v0.0.0-20201208040808-7e3f01d25324
-	google.golang.org/grpc v1.34.0
+	github.com/lib/pq v1.10.9
+	go.etcd.io/bbolt v1.3.7
+	golang.org/x/crypto v0.6.0
```
Comment (davecgh) [inline go.mod:27]:
```suggestion
	golang.org/x/crypto v0.10.0
```

File: pool/boltupgrades_test.go [hunk 7]
```diff
@@ -39,8 +39,12 @@ func TestBoltDBUpgrades(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	t.Cleanup(func() {
```
Comment (davecgh) [inline pool/boltupgrades_test.go:42]: I'd suggest just replacing all this and the `os.MkdirTemp` with the following which handles creation, cleanup, and errors:

```suggestion
	d:= t.TempDir()
```

File: run_tests.sh (hunk 3 above)
Comment (dajohi) [reply run_tests.sh:30]: @jholdstock says golangci-lint does a good  job at autodetection.

**Review by davecgh — APPROVED** [review]

**Review by jholdstock — COMMENTED** [review]
I have reviewed this without running it, everything looks fine to me.

The only hesitation I have is the blind bump from postgres 13 to 15. Whilst it is likely fine, I havent reviewed the postgres release notes and I dont know for sure that dcrpool is compatible.

**Pull request merged**